### PR TITLE
fix(metadata-editor): Fix error handling in metadata sidebar

### PR DESCRIPTION
**Pull request by jankowiakdawid**
## Description

QA reported we're showing wrong error notification when AI autofill action is run without internet connection. After accessing other tabs in sidebar we decided to show default error screen from `ErrorBoundry`.

I've also bumped version of `blueprint-web`, `blueprint-web-assets` and `metadata-editor` to get the new features in storybook.

As well as copied mocked responses from https://github.com/box/box-ui-elements/pull/3699

## Screenshot

Screenshot 2024-10-30 at 18 08 01

### CODE REVIEW

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx [hunk 1]
```diff
@@ -358,9 +388,86 @@ export const MetadataInstanceEditorAIEnabled: StoryObj<typeof MetadataSidebarRed
...
+            ...mockFeatures,
+            'metadata.aiSuggestions.enabled': true,
+        },
+    },
+    parameters: {
+        ...defaultVisualConfig.parameters,
+        msw: {
+            handlers: [
+                ...defaultMockHandlers,
+                http.post('https://api.box.com/2.0/ai/extract_structured', () => {
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx:410]: we should use the variable for our host here, in case if does change

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx [hunk 2]
```diff
@@ -358,9 +388,86 @@ export const MetadataInstanceEditorAIEnabled: StoryObj<typeof MetadataSidebarRed
...
+            dangerouslyIgnoreUnhandledErrors: true,
+        },
+    },
+    play: async ({ canvasElement }) => {
+        const canvas = within(canvasElement);
+
+        const autofillButton = await canvas.findByRole(
+            'button',
+            { name: 'Autofill My Template with Box AI' },
+            { timeout: 5000 },
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx:425]: do we really need 5s ? same for the other ones?

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]

File: src/api/Intelligence.js [hunk 3]
```diff
@@ -70,20 +69,13 @@ class Intelligence extends Base {
...
-            const { status } = e;
-            if (isUserCorrectableError(status)) {
-                throw e;
-            }
-        }
-        return getProp(suggestionsResponse, 'data', {});
+        const suggestionsResponse = await this.xhr.post({
+            url,
+            data: request,
+            id: `file_${request.items[0].id}`,
```
Comment (JChan106) [inline src/api/Intelligence.js:75]: Do we need to make sure request has the `items` field and that it is not empty?

File: package.json [hunk 4]
```diff
@@ -122,13 +122,13 @@
...
-        "@box/blueprint-web": "^7.30.1",
-        "@box/blueprint-web-assets": "^4.21.0",
+        "@box/blueprint-web": "^8.1.0",
+        "@box/blueprint-web-assets": "^4.28.0",
         "@box/box-ai-content-answers": "^0.57.1",
         "@box/cldr-data": "^34.2.0",
         "@box/frontend": "^10.0.0",
         "@box/languages": "^1.0.0",
-        "@box/metadata-editor": "^0.65.0",
+        "@box/metadata-editor": "^0.71.0",
```
Comment (tjuanitas) [inline package.json:131]: do we need to update peer dependency list?

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx (hunk 1 above)
Comment (jankowiakdawid) [reply src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx]: changed that to variable

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx (hunk 2 above)
Comment (jankowiakdawid) [reply src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx]: I've changed that to waiting for side panel to be available

File: package.json (hunk 4 above)
Comment (jankowiakdawid) [reply package.json:131]: Not necessarily as it takes latest from host app.
I'll do it either way and try to match versions we have currently in our app.

File: src/api/Intelligence.js (hunk 3 above)
Comment (jankowiakdawid) [reply src/api/Intelligence.js:75]: I've added the same checks as are in `ask` method

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx (hunk 2 above)
Comment (jankowiakdawid) [reply src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx]: Correction, I've changed timeout to `2000`, it works locally. 
Waiting for container was failing on re-runs.

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx [hunk 5]
```diff
@@ -41,7 +42,7 @@ export const AddTemplateDropdownMenuOn = {
     play: async ({ canvasElement }) => {
         const canvas = within(canvasElement);
 
-        const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' }, { timeout: 5000 });
+        const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' }, { timeout: 2000 });
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx:45]: so i was experimenting with this on your PR and I noticed this works, I think we should avoid using timeout to a hardcoded ms if possible
```
        waitFor(async () => {
            const addTemplateButton = await canvas.findByRole('button', { name: 'Add template' });
            expect(addTemplateButton).toBeInTheDocument();
            await userEvent.click(addTemplateButton);

            const customMetadataOption = await canvas.findByRole('option', { name: 'Virus Scan' });
            expect(customMetadataOption).toBeInTheDocument();
            await userEvent.click(customMetadataOption);

            const templateHeader = await canvas.findByRole('heading', { name: 'Virus Scan' });
            expect(templateHeader).toBeInTheDocument();
        });
```

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx (hunk 5 above)
Comment (jankowiakdawid) [reply src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx:45]: Hmmm, when I was running it like so I was running into timeouts.
lately they became so long that no story was working on Chrome for me. I've noticed that we have a call to global metadata templates unmocked. I've added it with empty array and it made all the stories much faster to load, almost instant.

I think the reason it worked for you and not us (in PL) was that unmocked calls were slower due to round trips.
Now it should be fixed.

File: src/elements/content-sidebar/stories/__mocks__/MetadataSidebarRedesignedMocks.ts [hunk 6]
```diff
@@ -1,13 +1,17 @@
-export const fileId = '415542803939';
+export const fileIdWithMetadata = '415542803939';
 export const fileIdWithoutMetadata = '416047501580';
 
+const host = 'https://api.box.com';
```
[the remark this answers is not in the document]
Comment (jankowiakdawid) [reply src/elements/content-sidebar/stories/__mocks__/MetadataSidebarRedesignedMocks.ts:4]: Sure thing! I wasn't aware of this constant

File: src/elements/content-sidebar/MetadataSidebarRedesign.tsx [hunk 7]
```diff
@@ -127,7 +127,11 @@ function MetadataSidebarRedesign({ api, elementId, fileId, onError, isFeatureEna
     };
 
     const handleDeleteInstance = async (metadataInstance: MetadataTemplateInstance) => {
-        await handleDeleteMetadataInstance(metadataInstance);
+        try {
+            await handleDeleteMetadataInstance(metadataInstance);
+        } catch {
+            // ignore error, handled in useSidebarMetadataFetcher
+        }
```
Comment (jankowiakdawid) [inline src/elements/content-sidebar/MetadataSidebarRedesign.tsx:134]: This fixed the `ShowErrorOnDelete` story on Safari for me.

Before this change the story was failing on Safari with all the steps passing due to uncaught error. I was able to narrow down it to this place.
Putting try catch inside `handleDeleteMetadataInstance` didn't fix the story. Put it around. We handle the errors with `onError` and `errorMessage` so no additional action is needed.

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]

**Review by greg-in-a-box — DISMISSED** [review]

File: src/api/Intelligence.js [hunk 8]
```diff
@@ -68,22 +67,26 @@ class Intelligence extends Base {
...
-            suggestionsResponse = await this.xhr.post({
-                url,
-                data: request,
-                id: `file_${request.items[0].id}`,
-            });
-        } catch (e) {
-            const { status } = e;
-            if (isUserCorrectableError(status)) {
-                throw e;
+        items.forEach(item => {
```
Comment (JChan106) [inline src/api/Intelligence.js:75]: Do we need a forEach here? I see we only use items[0] for the `suggestionsResponse`. If so, can we just throw an error only if `!items[0].id`?

File: src/elements/content-sidebar/hooks/useSidebarMetadataFetcher.ts [hunk 9]
```diff
@@ -188,18 +195,28 @@ function useSidebarMetadataFetcher(
         [api, file, onApiError],
     );
 
+    const [, setError] = React.useState();
```
[the remark this answers is not in the document]
Comment (jankowiakdawid) [reply src/elements/content-sidebar/hooks/useSidebarMetadataFetcher.ts:198]: It's explained at the usage of `setState`
```
// react way of throwing errors from async callbacks - https://github.com/facebook/react/issues/14981#issuecomment-468460187
setError(() => {
    throw error;
});
```

**tl;dr** this is the React way of throwing errors in async callbacks

**Review by JChan106 — APPROVED** [review]

**Review by greg-in-a-box — APPROVED** [review]
